### PR TITLE
fix: Only look at kubeclientset pods if needed

### PR DESCRIPTION
**Pull request by tooptoop4**

### CODE REVIEW

**Review by LinuxSuRen — APPROVED** [review]
LGTM

**Comment by LinuxSuRen** [comment]
hi @tooptoop4 I guess you might need to keep the PR template instead of deleting it.

**Review by alexec — CHANGES_REQUESTED** [review]
Looks good, but can you:

* Sign-off your commits to the DCO check passes.
* Add comments saying why this is a good idea.

**Comment by tooptoop4** [comment]
crunchety

File: workflow/controller/controller.go [hunk 1]
```diff
@@ -502,7 +502,6 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 	logCtx := log.WithFields(log.Fields{"key": key, "action": action})
 	logCtx.Info("cleaning up pod")
 	err := func() error {
-		pods := wfc.kubeclientset.CoreV1().Pods(namespace)
```
Comment (terrytangyuan) [inline workflow/controller/controller.go:505]: This doesn't actually look up pods, it only invokes the API. The patch/delete calls are the actual calls to K8s. I don't see any benefits this brings.

**Pull request closed without merging**